### PR TITLE
Feat/better error messages

### PR DESCRIPTION
**Pull request by Albertlarsen**
## Description
adds better Error messages and error handling in frontend

## Related Issue(s)
- #372 
- #385 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/components/ErrorPanel/ErrorPanel.tsx
Comment (allinox) [inline src/components/ErrorPanel/ErrorPanel.tsx:1]: Ikke relatert til noen spesifikk fil egentlig, men jeg prøvde å teste dette lokalt ved å fremprovosere en 500-feil, og da krasjet siden 😅

image

File: src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts [hunk 1]
```diff
@@ -322,7 +338,14 @@ const overviewOrgSlice = createSlice({
       })
       .addCase(fetchOverviewOrgsOffered.rejected, (state, action) => {
         state.loading = true;
-        state.error = action.error.message ?? 'Unknown error';
+        state.error.code = String(action.payload?.response?.status) ?? 'Unknown code';
```
Comment (allinox) [inline src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts:341]: Did you mean `state.error.statusCode` here? (And in the other refs to it?)
Maybe that's what makes it crash?

Comment (Albertlarsen) [reply src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts]: Yup, good catch! Changed it now

File: src/components/ErrorPanel/ErrorPanel.tsx
Comment (Albertlarsen) [reply src/components/ErrorPanel/ErrorPanel.tsx:1]: Kan hende det var fordi det du kommenterte med code vs statusCode. Jeg får i alle fall opp 500-error. Men prøv å bare stoppe backend(ikke bff) og reload siden. Da skal feilen bli som under der.

image

**Review by allinox — CHANGES_REQUESTED** [review]
A lot of good here!
But the page still crashes for me, so that should probably be fixed 😅

File: src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts [hunk 2]
```diff
@@ -152,50 +158,54 @@ export const fetchOverviewOrgsReceived = createAsyncThunk(
...
+              ],
+            },
+          ],
+        },
+      );
+
+      return response.data;
+    } catch (error) {
+      console.error(error);
+      return rejectWithValue(error);
```
[the remark this answers is not in the document]
Comment (Albertlarsen) [reply src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts:201]: rejectWithValue sends all information (error code, message) to the rejected addCase. throw Error only sends either code or message, not both

[the remark this answers is not in the document]
Comment (allinox) [reply src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts:201]: Ahh, ok 👍

File: src/components/ErrorPanel/ErrorPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  const time = `${currentDate.getHours()}:${currentDate.getMinutes()}:${currentDate.getSeconds()}`;
+
+  return (
+    <Panel
+      title={title}
+      variant={PanelVariant.Error}
+      forceMobileLayout
+    >
+      <div>
+        <List borderStyle='dashed'>
```
[the remark this answers is not in the document]
Comment (Albertlarsen) [reply src/components/ErrorPanel/ErrorPanel.tsx]: I agree and I'll fix it :)

File: src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts [hunk 4]
```diff
@@ -134,7 +134,7 @@ export const fetchOverviewOrgsOffered = createAsyncThunk(
       }
 
       const response = await axios.get(
-        `/accessmanagement/api/v1/${altinnPartyId}/maskinportenschema/offered`,
+        `/accessmanagement/api/v1/${altinnPartyId}2133/maskinportenschema/offered`,
```
Comment (allinox) [inline src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts:137]: What's this added number?

File: src/components/ErrorPanel/ErrorPanel.tsx
Comment (allinox) [reply src/components/ErrorPanel/ErrorPanel.tsx:1]: Now it works, and looks great to boot! 😄
image

File: src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts (hunk 4 above)
Comment (Albertlarsen) [reply src/rtk/features/apiDelegation/overviewOrg/overviewOrgSlice.ts:137]: Woops, a number to provoke 400-error 🥲

**Review by allinox — CHANGES_REQUESTED** [review]
Looks good 😊
Only thing left is to remove the numbers provoking the 400 error mentioned in my final comment

**Review by allinox — APPROVED** [review]
Awesome sauce 😎

**Pull request merged**